### ISSUE TITLE
Course model

### ISSUE THREAD

**Issue by SkaeX** [BrainMaestro/FCE#171]
I think we should have a course model. Thoughts @BrainMaestro, @tenorz007

**Comment by tenorz007** [comment]
For what reason?

**Comment by SkaeX** [comment]
@tenorz007 Naturally, a course has many sections. We omitted that. That is why we have repetition of course title, course code,

**Comment by tenorz007** [comment]
Repetition where! What does the current schema look like.

**Comment by SkaeX** [comment]
```
$table->string('course_code');
$table->unsignedInteger('semester_id');
$table->unsignedInteger('school_id')->index();
$table->string('course_title');
```
For `X` semesters we should expect appoximately `X` `course_code` and `X` `course_title`

**Comment by tenorz007** [comment]
This way courses are different over semesters aren't they

**Comment by SkaeX** [comment]
Courses repeat over semesters.

**Comment by tenorz007** [comment]
Do course codes remain the same over semesters?

**Comment by SkaeX** [comment]
Course codes are tied to courses not sections, so they repeat.

**Comment by tenorz007** [comment]
Repeat in what sense, does the course code change by section or something.

**Comment by SkaeX** [comment]
Even in a semester there can be 2 sections of MAT 101.

**Comment by tenorz007** [comment]
Would they have different course codes?

**Comment by SkaeX** [comment]
No, same title and course code.

**Comment by tenorz007** [comment]
But that can be handled using section since it's basically a different
section.

**Comment by SkaeX** [comment]
I don't understand. Please elaborate on how.

**Comment by tenorz007** [comment]
If a course has more than one section it's basically a new section since
everything about it is different except the course. Or am I failing to
understand your point!

**Comment by SkaeX** [comment]
Since the course is the same, course properties like `course_code` and `course_title` repeat.

**Comment by tenorz007** [comment]
Yeah I know, but isn't that the same with the way it's listed on open erp.

**Comment by BrainMaestro** [comment]
Yeah @tenorz007 it's the same way it's listed on open erp. Also crn differentiates the different sections. Is the duplication a lot?

**Comment by SkaeX** [comment]
Well, the duplication will grow with time. Since for each semester we intend to repeat many courses.
